apps/jubmoji-quest/src/components/LeaderBoard.tsx

```diff
@@ -37,37 +68,45 @@ export default function LeaderBoard() {
           <RankLabel variant="primary">Name</RankLabel>
           <RankLabel variant="primary">Score</RankLabel>
         </LeaderBoardContent>
-        <LeaderBoardContent>
-          <RankLabel>Rank</RankLabel>
-          <RankLabel>Name</RankLabel>
-          <RankLabel>Score</RankLabel>
-        </LeaderBoardContent>
-        <LeaderBoardContent>
-          <RankLabel>Rank</RankLabel>
-          <RankLabel>Name</RankLabel>
-          <RankLabel>Score</RankLabel>
-        </LeaderBoardContent>
-        <LeaderBoardContent>
-          <RankLabel variant="active">Rank</RankLabel>
-          <RankLabel variant="active">Name</RankLabel>
-          <RankLabel variant="active">Score</RankLabel>
-        </LeaderBoardContent>
-        <LeaderBoardContent>
-          <RankLabel>Rank</RankLabel>
-          <RankLabel>Name</RankLabel>
-          <RankLabel>Score</RankLabel>
-        </LeaderBoardContent>
-        <LeaderBoardContent>
-          <RankLabel>Rank</RankLabel>
-          <RankLabel>Name</RankLabel>
-          <RankLabel>Score</RankLabel>
-        </LeaderBoardContent>
-        <LeaderBoardContent>
-          <RankLabel>Rank</RankLabel>
-          <RankLabel>Name</RankLabel>
-          <RankLabel>Score</RankLabel>
-        </LeaderBoardContent>
+        {loading ? (
+          <>
+            <LoadingContent />
+            <LoadingContent />
+            <LoadingContent />
+            <LoadingContent />
+          </>
+        ) : hasRanking ? (
+          <div className="overflow-scroll max-h-40 pr-1">
+            <div className="flex flex-col gap-2 w-full">
+              {ranking?.map(([pubKeyIndex, score], index) => {
+                const rank = index + 1;
+                const card = getJubmojiCardByPubIndex(
+                  jubmojiCollectionCards,
+                  Number(pubKeyIndex)
+                );
+
+                return (
+                  <LeaderBoardContent key={pubKeyIndex}>
+                    <RankLabel>{rank}</RankLabel>
+                    <RankLabel>
+                      <span>
+                        {card?.emoji} {card?.name}
+                      </span>
+                    </RankLabel>
+                    <RankLabel>{score}</RankLabel>
+                  </LeaderBoardContent>
+                );
+              })}
+            </div>
+          </div>
+        ) : (
+          <Message className="mx-auto"> No scores yet</Message>
+        )}
       </div>
     </Card.Base>
   );
-}
+};
+
+LeaderBoard.displayName = "LeaderBoard";
```

Comment:
Can we use consistent casing here or is this another weird file name issue

Reply:
oh just have `LeaderBoard` everywhere, you suggest to change to `Leaderboard`?

Reply:
its a super minor nit, just to have the naming match the hook & backend. But if it takes any significant amount of time, feel free to ignore